src/view/xwayland.cpp

```diff
@@ -400,7 +438,10 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
         }
 
         if (xw->fullscreen)
-            fullscreen_request(get_output(), true);
+        {
+                this->view_impl->last_windowed_geometry = get_output_geometry();
```

Comment:
1) output geometry is the wrong thing to take, you should pick wm_geometry
2) what bug(s) does this fix? This should already be handled in fullscreen_request

Reply:
none really as mentioned above a desperate try to see why swticher crashes.

Reply:
alright, then, since this doesn't change anything, revert the change, most probably the bug is in switcher itself.